kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -215,16 +222,31 @@ private void commitToTable(
       LOG.info("Nothing to commit to table {}, skipping", tableIdentifier);
     } else {
       if (deleteFiles.isEmpty()) {
-        AppendFiles appendOp = table.newAppend();
-        branch.ifPresent(appendOp::toBranch);
-        appendOp.set(snapshotOffsetsProp, offsetsJson);
-        appendOp.set(COMMIT_ID_SNAPSHOT_PROP, commitState.currentCommitId().toString());
-        if (vtts != null) {
-          appendOp.set(VTTS_SNAPSHOT_PROP, Long.toString(vtts));
-        }
-        dataFiles.forEach(appendOp::appendFile);
-        appendOp.commit();
+        Map<Integer, List<DataFile>> dataFilesBySpec = new HashMap<>();
+        dataFiles.forEach(dataFile -> {
+          List<DataFile> list = dataFilesBySpec.computeIfAbsent(dataFile.specId(), ignored -> Lists.newArrayList());
+          list.add(dataFile);
+        });
+        Integer maxSpecId = dataFilesBySpec.keySet().stream().max(Integer::compare).get();
+
+        Transaction transaction = table.newTransaction();
+        dataFilesBySpec.entrySet().stream().sorted(Comparator.comparingInt(Map.Entry::getKey)).forEach(entry -> {
+          AppendFiles appendOp = transaction.newAppend();
+          branch.ifPresent(appendOp::toBranch);
+          entry.getValue().forEach(appendOp::appendFile);
+          appendOp.set(COMMIT_ID_SNAPSHOT_PROP, commitState.currentCommitId().toString());
```

Comment:
Note however that we include the commit-id snapshot property on each snapshot we add as part of this transaction. We don't have to do this but it could be helpful for debugging purposes and it's easy enough to do.

Reply:
So each append gets the same metadata but we don't really care.